src/skip.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A predicate that determines if the given name should be skipped
+ * in renaming or disallowed to be produced by a renaming function
+ * @see strategy.ts
+ */
+export type SkipPredicate = (name: string) => boolean;
+
+/**
+ * Creates a SkipPredicate that filters against the given set of strings or
+ * regexes.
+ * @param except
+ * @returns SkipPredicate
+ */
+export function createSkip(except?: Iterable<string | RegExp>): SkipPredicate {
+  if (except === undefined) {
```

Comment:
Nit: Using `===` may be too strict and it could miss the odd case where `null` is passed. It's generally safer to use `==` when comparing null or undefined, and it may be more idiomatic (shorter too) in this case to check for falsyness instead. For example:

```suggestion
  if (!except) {
```

Reply:
Hmm, I was mostly working with the assumption that the type system would catch cases like `createSkip(null)` (which is a type error), though I guess when we import this, types are completely erased so we lose this safety

Reply:
Users could always self-sabotage and use `null as any` 😅

It doesn't matter too much for the internal fork because we ban `as any` or other bad styles, but it's better to be a bit safer when possible for users who import this library without strict types